### PR TITLE
Adds support for event add-handler accessors.

### LINKED ISSUE #6313: VBCSCompiler crash with any custom WinRT events in VB
REPRO: (this is the simplest example of a custom WinRT event that I could come up with)

(1) File>New>VB>Windows>Universal>BlankApp

(2) Paste the following code
```vb
Public NotInheritable Class C
    Implements IBackgroundTaskInstance

    Public Property InstanceId As Guid Implements IBackgroundTaskInstance.InstanceId
    Public Property Progress As UInteger Implements IBackgroundTaskInstance.Progress
    Public Property SuspendedCount As UInteger Implements IBackgroundTaskInstance.SuspendedCount
    Public Property Task As BackgroundTaskRegistration Implements IBackgroundTaskInstance.Task
    Public Property TriggerDetails As Object Implements IBackgroundTaskInstance.TriggerDetails
    Public Function GetDeferral() As BackgroundTaskDeferral Implements IBackgroundTaskInstance.GetDeferral
        Return Nothing
    End Function

    Private CanceledEvent As EventRegistrationTokenTable(Of BackgroundTaskCanceledEventHandler)

    Public Custom Event Canceled As BackgroundTaskCanceledEventHandler Implements IBackgroundTaskInstance.Canceled
        AddHandler(handler As BackgroundTaskCanceledEventHandler)
            Return EventRegistrationTokenTable(Of BackgroundTaskCanceledEventHandler).
                GetOrCreateEventRegistrationTokenTable(CanceledEvent).
                AddEventHandler(handler)
        End AddHandler
        RemoveHandler(token As EventRegistrationToken)
            CanceledEvent?.RemoveEventHandler(token)
        End RemoveHandler
        RaiseEvent(sender As IBackgroundTaskInstance, reason As BackgroundTaskCancellationReason)
            CanceledEvent?.InvocationList.Invoke(sender, reason)
        End RaiseEvent
    End Event

    Sub Trigger(reason As BackgroundTaskCancellationReason)
        RaiseEvent Canceled(Me, reason)
    End Sub

End Class
```

(3) Add the required Imports statement, and build the project.

WHAT I GET: During build, "VBCSCompiler has stopped working. Windows is sending the information to Microsoft..." and then "vbc.exe has stopped working..."

WHAT I EXPECT: It should build fine. Note that the corresponding C# code builds fine...

```cs
public sealed class C : IBackgroundTaskInstance
{
    public BackgroundTaskDeferral GetDeferral() => null;
    public Guid InstanceId { get; set; }
    public uint Progress { get; set; }
    public uint SuspendedCount { get; set; }
    public BackgroundTaskRegistration Task { get; set; }
    public object TriggerDetails { get; set; }

    private EventRegistrationTokenTable<BackgroundTaskCanceledEventHandler> CanceledEvent;

    public event BackgroundTaskCanceledEventHandler Canceled
    {
        add
        {
            return EventRegistrationTokenTable<BackgroundTaskCanceledEventHandler>
                .GetOrCreateEventRegistrationTokenTable(ref CanceledEvent)
                .AddEventHandler(value);
        }
        remove
        {
            CanceledEvent?.RemoveEventHandler(value);
        }
    }

    public void Trigger(BackgroundTaskCancellationReason reason)
    {
        CanceledEvent?.InvocationList.Invoke(this, reason);
    }

}
```

### PR DESCRIPTION
**Pull request by TyOverby**
Closes #6313

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Symbols/Source/LocalSymbol.vb [hunk 1]
```diff
@@ -501,6 +501,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                             Return propertyBlock.Accessors.Where(Function(a) a.IsKind(SyntaxKind.GetAccessorBlock)).Single().BlockStatement
                         End If
 
+                        If node.IsKind(SyntaxKind.EventStatement) Then
```
Comment (TyOverby) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/LocalSymbol.vb]: What does the style guide say about this?  Should this be an ElseIf?

**Comment by TyOverby** [comment]
@tmat: this is modifying some code that you added back in 916b9efafe3bc1544430388a59e4dea523f9880b.  Could you review?

**Comment by TyOverby** [comment]
@AlekseyTs I managed to get a test case that failed before the fix, and works afterwards.

**Comment by AlekseyTs** [comment]
CC @dotnet/roslyn-compiler

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb [hunk 2]
```diff
@@ -896,6 +896,35 @@ End Class
 ]]>)
         End Sub
 
+        <Fact()>
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb]: WorkItem?

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb [hunk 3]
```diff
@@ -896,6 +896,35 @@ End Class
...
+        RaiseEvent()
+
+        End RaiseEvent
+    End Event
+End Class
+    </file>
+</compilation>
+
+            Dim comp = CreateCompilationWithReferences(source, WinRtRefs, options:=TestOptions.DebugWinMD)
+            Dim verifier = CompileAndVerify(comp)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb]: Consider adding verification that emitted PDB information makes sense. @tmat Should be able to help with that.

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb (hunk 2 above)
Comment (TyOverby) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb]: Done

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb (hunk 3 above)
Comment (TyOverby) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenWinMdEvents.vb]: @tmat Thinks that using winrt disables ENC.

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler Review please!

**Comment by VSadov** [comment]
LGTM

**Comment by TyOverby** [comment]
Closed in favor of #6448

**Comment by MattGertz** [comment]
Approved.

**Pull request closed without merging**